ethereum-explorer/src/map_filter_transactions.rs

```diff
@@ -0,0 +1,77 @@
+use substreams_ethereum::pb::eth::v2::{Block, TransactionTraceStatus};
+use crate::pb::eth::transaction::v1::{Transaction, Transactions};
+use serde::Deserialize;
+use substreams::errors::Error;
+use crate::util;
+
+#[derive(Deserialize)]
+struct TransactionFilterParams {
+    hash: Option<String>,
+    to: Option<String>,
+    from: Option<String>
+}
+
+#[substreams::handlers::map]
+pub fn map_filter_transactions(params: String, blk: Block) -> Result<Transactions, Vec<substreams::errors::Error>> {
+    let filters: TransactionFilterParams = serde_qs::from_str(&params).unwrap();
+    let errors = verify_filter_params(&filters);
+    if errors.len() > 0 {
+        return Err(errors)
+    }
+
+    let mut filtered_transactions: Vec<Transaction> = Vec::new();
+    
+    for transaction in &blk.transaction_traces {
+        let tx_hash = util::hexadecimal_to_string(&transaction.hash);
```

Comment:
I still personally think `hash` is really not a good showcase. I leave the decision to you to include it or not.